zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchClientIT.java

```diff
@@ -14,83 +14,52 @@
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
 import java.util.List;
 import java.util.Map;
-import java.util.UUID;
-import org.agrona.CloseHelper;
+import java.util.Optional;
 import org.assertj.core.api.InstanceOfAssertFactories;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.opensearch.testcontainers.OpensearchContainer;
-import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-@Testcontainers
-final class OpensearchClientIT {
-  private static final String PASSWORD = "P@a$5w0rd";
-  private static final String ADMIN_PASSWORD_ENV_VAR = "OPENSEARCH_INITIAL_ADMIN_PASSWORD";
+public class OpensearchClientIT {
 
-  @Container
-  private static final OpensearchContainer<?> CONTAINER =
-      TestSearchContainers.createDefaultOpensearchContainer()
-          .withSecurityEnabled()
-          .withEnv(ADMIN_PASSWORD_ENV_VAR, PASSWORD);
+  protected static final String IT_OPENSEARCH_AWS_INSTANCE_URL_PROPERTY =
+      "camunda.it.opensearch.aws_instance_url";
 
   private static final int PARTITION_ID = 1;
 
-  private final ProtocolFactory recordFactory = new ProtocolFactory();
-  private final OpensearchExporterConfiguration config = new OpensearchExporterConfiguration();
-  private final TemplateReader templateReader = new TemplateReader(config.index);
-  private final RecordIndexRouter indexRouter = new RecordIndexRouter(config.index);
-  private final BulkIndexRequest bulkRequest = new BulkIndexRequest();
-
-  private TestClient testClient;
-  private OpensearchClient client;
-
-  @BeforeEach
-  public void beforeEach() {
-    // as all tests use the same endpoint, we need a per-test unique prefix
-    config.index.prefix = UUID.randomUUID() + "-test-record";
-    config.url = CONTAINER.getHttpHostAddress();
-    config.getAuthentication().setUsername(CONTAINER.getUsername());
-    config.getAuthentication().setPassword(PASSWORD);
-    testClient = new TestClient(config, indexRouter);
-    client =
-        new OpensearchClient(
-            config,
-            bulkRequest,
-            RestClientFactory.of(config, true),
-            indexRouter,
-            templateReader,
-            new OpensearchMetrics(new SimpleMeterRegistry()));
-  }
+  @RegisterExtension private static SearchDBExtension searchDB = initializeSearchDBExtension();
 
-  @AfterEach
-  void afterEach() {
-    CloseHelper.quietCloseAll(testClient, client);
+  private static SearchDBExtension initializeSearchDBExtension() {
+    final var openSearchAwsInstanceUrl =
+        Optional.ofNullable(System.getProperty(IT_OPENSEARCH_AWS_INSTANCE_URL_PROPERTY)).orElse("");
+    if (openSearchAwsInstanceUrl.isEmpty()) {
+      return new ContainerizedSearchDBExtension();
+    } else {
+      return new AWSSearchDBExtension(openSearchAwsInstanceUrl);
+    }
   }
```

Comment:
🔧 I think I would make this factory method part of the SearchDBextension. 

Then like:

```
@RegisterExtension private static SearchDBExtension searchDB = SearchDBExtension.create();
```

Reply:
Fair point

Reply:
Changes moved here https://github.com/camunda/camunda/pull/24986